### PR TITLE
chore: Bumps version 2.3.0-beta.1 and update CHANGELOG.

### PR DESCRIPTION
**Pull request by SandPod**
Bumps the version to `2.3.0-beta.1` and updates the CHANGELOG.

## Pre-launch Checklist

- [ ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

_If you have done any breaking changes, make sure to outline them here, so that they can be included in the notes for the next release._

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+## 2.3.0-beta.1
+- feat: Add support for transaction isolation levels.
+- feat: Adds typed interface for transaction savepoints.
+- feat: Adds support for endpoint definition placement anywhere in server's `src` directory.
+- feat: Adds support for model definitions placement anywhere in server's `src` directory.
+- fix: Improve message transmission guarantee in method streams.
+- chore: Update `vm_service` dependency range.
```
Comment (vlidholt) [inline CHANGELOG.md:7]: We typically don't include chore/refactor in the changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,17 @@
+## 2.3.0-beta.1
+- feat: Add support for transaction isolation levels.
```
Comment (vlidholt) [inline CHANGELOG.md:2]: For consistency:

Adds support for transaction isolation levels.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+- feat: Adds typed interface for transaction savepoints.
+- feat: Adds support for endpoint definition placement anywhere in server's `src` directory.
+- feat: Adds support for model definitions placement anywhere in server's `src` directory.
+- fix: Improve message transmission guarantee in method streams.
```
Comment (vlidholt) [inline CHANGELOG.md:6]: Improves

**Review by vlidholt — APPROVED** [review]
LGTM with comments fixed. :)

File: CHANGELOG.md (hunk 1 above)
Comment (SandPod) [reply CHANGELOG.md:7]: Aaah, I found it other changelogs and was uncertain. Will remove it!

**Pull request merged**